src/api/app/controllers/trigger_controller.rb

```diff
@@ -1,53 +1,96 @@
 class TriggerController < ApplicationController
+  validate_action rebuild: { method: :post, response: :status }
+  validate_action release: { method: :post, response: :status }
   validate_action runservice: { method: :post, response: :status }
 
+  before_action :extract_auth_from_request, :validate_auth_token, :require_valid_token
+
+  before_action :require_project_param, only: [:release]
   #
-  # This controller is checking permission always only on the base of tokens
+  # Authentication happens with tokens, so no login is required
   #
   skip_before_action :extract_user
   skip_before_action :require_login
 
-  # github.com sends a hash payload
-  skip_before_action :validate_params, only: [:runservice]
+  def rebuild
+    Backend::Api::Sources::Package.rebuild(@pkg.project.name, @pkg.name)
+    render_ok
+  end
 
-  def runservice
-    auth = request.env['HTTP_AUTHORIZATION']
+  # rubocop:disable Lint/NonLocalExitFromIterator
+  def release
+    matched_repo = false
+    User.session = @token.user
```

Comment:
IMHO we should not do this, because any programming error or refactoring may lead to a security issue.
The token should only allow a certain operation, but we should prepare for side effects.

The problem coolo reported is actually a generic problem that we always try to write the package meta when using Package.source_changed. source_changed was supposed just to update database content which could get replayed from the source server.
However, this is beyond the topic of this pull request and a generic problem atm.

Reply:
I think the issue isn't about the controller, but operations on models and backend that need an user.. I think thats horrible too, but I'm not sure we are able to avoid it now.

Reply:
create a backlog item and continue :(